Tasks/Gradle/CodeAnalysis/Common/BaseTool.ts

```diff
@@ -0,0 +1,153 @@
+import {AnalysisResult} from './AnalysisResult'
+import {IAnalysisTool} from './IAnalysisTool'
+import {BuildOutput, BuildEngine} from './BuildOutput'
+import {ModuleOutput} from './ModuleOutput'
+import {ToolRunner} from 'vsts-task-lib/toolrunner';
+
+import path = require('path');
+import fs = require('fs');
+import glob = require('glob');
+import xml2js = require('xml2js');
+
+import tl = require('vsts-task-lib/task');
+
+
+/**
+ * An abstract class that is the base for both configuring a build to use an analysis tool and   
+ * 
+ * @export
+ * @class PmdReportParser
+ * @implements {IAnalysisToolReportParser}
+ */
+export abstract class BaseTool implements IAnalysisTool {
```

Comment:
This is pretty much existing code - I refactored the logic from PmdTool into an abstract class to share it with the CheckstyleTool. Tools are responsible for both configuring the build and parsing the xml results.